[jdbc] Update dependencies

lsiepel opened the pull request:
Upgrade all jdbc persistence library's to the most recent.

Bump com.h2database:h2 from 2.2.224 from to 2.3.232

Bump org.hsqldb:hsqldb from 2.3.3 to 2.7.4

Bump org.mariadb.jdbc:mariadb-java-client from 3.0.8 to 3.5.5
- [Commits](https://github.com/mariadb-corporation/mariadb-connector-j/compare/3.0.8...3.5.5)

Bumps [com.mysql:mysql-connector-j](https://github.com/mysql/mysql-connector-j) from 9.2.0 to 9.4.0.
- [Changelog](https://github.com/mysql/mysql-connector-j/blob/release/9.x/CHANGES)
- [Commits](https://github.com/mysql/mysql-connector-j/compare/9.2.0...9.4.0)

Bump org.postgresql:postgresql from 42.4.4 to 42.7.7 3.0.8

Bump org.xerial:sqlite-jdbc 3.42.0.0 to 3.50.3.0

I also tried to upgrade com.oracle.database.jdbc.ojdbc11 from 23.5.0.2407 to 23.9.0.25.07 and upgrade to ojdbc17. But it seemed more difficult then expected due to the oracle lib needed to be osgified. Not sure if that is still the case. Hope @mherwege can have a look to update lib.

@florian-h05 i guess the files need to be uploaded to jfrog. Please wait before all is approved to prevent unneeded work.

mherwege commented:
For Oracle, when I added it last year, the maven artifacts from Oracle did not have the required info for use in an OSGI environment out of the box, so I had to go through osgifying them. I don’t think that changed.
The last 2 segments of the version where collapsed when osgifying because the OSGI spec only allows 4 segments in a version, and it didn’t resolve without doing that.

lsiepel commented:
> For Oracle, when I added it last year, the maven artifacts from Oracle did not have the required info for use in an OSGI environment out of the box, so I had to go through osgifying them. I don’t think that changed. The last 2 segments of the version where collapsed when osgifying because the OSGI spec only allows 4 segments in a version, and it didn’t resolve without doing that.

Ah thanks, i just reverted oracle before reading your comment. Oracle now also has a new package, so maybe it improved.

mherwege commented:
> Oracle now also has a new package, so maybe it improved.

There is an update about every month. But I don’t think it fundamentally changes the bundle info they include. It has been like that for ever. So if we want to upgrade the version, we need to create a new osgifyed bundle and test.

jlaur reviewed (changes requested):
Can you also update the README and `JdbcConfiguration`? See #18474.

I will give the new MySQL Connector/J a test.

lsiepel commented:
> @lsiepel Just uploaded all required artefacts to JFrog before I leave for vacation

Hope you will have a good time, enjoy!

jlaur reviewed (approved):
Thanks, let's get it in before first milestone for some test mileage, hopefully.
